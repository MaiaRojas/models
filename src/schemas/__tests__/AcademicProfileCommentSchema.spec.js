const mongoose = require('mongoose/browser');
const AcademicProfileCommentSchema = require('../AcademicProfileCommentSchema')(mongoose);
const CohortProjectSchema = require('../CohortProjectSchema')(mongoose);
const UserSchema = require('../UserSchema')(mongoose);

describe('AcademicProfileCommentSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, AcademicProfileCommentSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const createdBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);
    const cohortProject = new mongoose.Document({}, CohortProjectSchema);

    const doc = new mongoose.Document({
      createdBy: createdBy._id,
      cohortProject: cohortProject._id,
      type: 'tech',
      text: 'foo',
    }, AcademicProfileCommentSchema);

    doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
