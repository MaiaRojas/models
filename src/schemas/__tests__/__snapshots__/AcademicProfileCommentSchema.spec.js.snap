// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileCommentSchema should fail validation when missing fields are provided 1`] = `
Object {
  "cohortProject": [ValidatorError: Path \`cohortProject\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "text": [ValidatorError: Path \`text\` is required.],
  "type": [ValidatorError: Path \`type\` is required.],
}
`;
